packages/circuit-ui/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,258 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use client';
+
+import {
+  flip,
+  offset as offsetMiddleware,
+  type Placement,
+  size,
+  type SizeOptions,
+  useFloating,
+} from '@floating-ui/react-dom';
+import {
+  type ComponentType,
+  forwardRef,
+  Fragment,
+  type KeyboardEvent,
+  useCallback,
+  useEffect,
+  useId,
+  useRef,
+  useState,
+} from 'react';
+
+import { Dialog, type DialogProps } from '../Dialog/Dialog.js';
+import type { ClickEvent } from '../../types/events.js';
+import { useStackContext } from '../StackContext/index.js';
+import { useMedia } from '../../hooks/useMedia/index.js';
+import { sharedClasses } from '../../styles/shared.js';
+import { applyMultipleRefs } from '../../util/refs.js';
+import { clsx } from '../../styles/clsx.js';
+import { usePrevious } from '../../hooks/usePrevious/index.js';
+
+import classes from './Popover.module.css';
+
+export interface PopoverReferenceProps {
+  'onClick': (event: ClickEvent) => void;
+  'onKeyDown'?: (event: KeyboardEvent) => void;
+  'id': string;
+  'aria-haspopup': boolean;
+  'aria-controls': string;
+  'aria-expanded': boolean;
+}
+
+export interface PopoverProps
+  extends Omit<
+    DialogProps,
+    | 'open'
+    | 'onCloseEnd'
+    | 'onCloseStart'
+    | 'isModal'
+    | 'animationDuration'
+    | 'preventClose'
+    | 'initialFocusRef'
+    | 'preventOutsideClickRefs'
+  > {
+  /**
+   * Determines whether the Popover is open or closed.
+   */
+  isOpen: boolean;
+  /**
+   * Function that is called when the Popover is closed.
+   */
+  onClose: DialogProps['onCloseEnd'];
+  /**
+   * One of the accepted placement values.
+   * @default `bottom`.
+   */
+  placement?: Placement;
+  /**
+   * The placements to fallback to when there is not enough space for the
+   * Popover.
+   * @default `['top', 'right', 'left']`.
+   */
+  fallbackPlacements?: Placement[];
+  /**
+   * Displaces the floating element from its `placement` along specified axes.
+   *
+   * Pass a number to move the floating element on the main axis, away from (if
+   * positive) or towards (if negative) the reference element. Pass an object
+   * to displace the floating element on both the main and cross axes.
+   */
+  offset?: number | { mainAxis?: number; crossAxis?: number };
+  /**
+   * The component that toggles the Popover when clicked. Also referred to as
+   * reference element.
+   */
+  component: ComponentType<PopoverReferenceProps>;
+}
+
+const sizeOptions: SizeOptions = {
+  apply({ availableHeight, elements }) {
+    elements.floating.style.setProperty(
+      '--popover-max-height',
+      `${availableHeight}px`,
+    );
+  },
+};
+
+export const Popover = forwardRef<HTMLDialogElement, PopoverProps>(
+  (
+    {
+      isOpen: initialOpen = false,
```

Comment:
What do you think about calling this `defaultOpen` to make it clear that the prop only sets the initial state and has no effect when updated? This would make it consistent with the Toggletip component.